ci/test/cargo-test/image/run-tests

```diff
@@ -20,7 +20,30 @@ ci_init
 
 while read -r binary cwd; do
     cd "$cwd"
-    ci_try "/tests/$binary" --nocapture
+    ci_try "/tests/$binary" --nocapture -Z unstable-options --format json --report-time | tee results.json
+    ci_try "/usr/local/cargo2junit" < results.json > junit.xml
+    ci_try "/bin/cat" junit.xml | sed 's/"/\\"/g' > escaped.xml
+    junit=$(<escaped.xml)
+    curl --request POST \
+         --url https://analytics-api.buildkite.com/v1/uploads \
+         --header 'Authorization: Token token="9uPd3ZeKfYQ2cKkEyGo1vaPU";' \
+         --header 'Content-Type: application/json' \
+         --data @- << EOF
+            {
+                "format": "junit",
+                "run_env": {
+                    "CI": "buildkite",
+                    "key": "BUILDKITE_BUILD_ID",
+                    "number": "BUILDKITE_BUILD_NUMBER",
+                    "job_id": "BUILDKITE_JOB_ID",
+                    "branch": "BUILDKITE_BRANCH",
+                    "commit_sha": "BUILDKITE_COMMIT",
+                    "message": "BUILDKITE_MESSAGE",
+                    "url": "BUILDKITE_BUILD_URL"
+                },
+                "data": "$junit"
+            }
+EOF
```

Comment:
and then all of this can get pushed into Python:

```py
from materialize import ROOT
junit_xml = spawn.capture(["cargo2junit", ROOT / "results.json"])
requests.post(
    "https://analytics-api.buildkite.com/v1/uploads",
    headers={"Authorization": f"Token {os.environ('BUILDKITE_TEST_ANALYTICS_API_KEY_CARGO_TEST'])}"},
    json={
        "format": "junit",
        "run_env": {
            "key": os.environ["BUILDKITE_BUILD_ID"],
            # ...
        },
        "data": junit_xml,
    }
)
```

You'll also need to make sure all of those env vars get forwarded into the ci-builder here:

https://github.com/MaterializeInc/materialize/blob/d6329387a092caad3654e9756d1f8847d0345fdb/bin/ci-builder#L144-L154

including the `BUILDKITE_TEST_ANALYTICS_API_KEY_CARGO_TEST` env var that I just added!

Reply:
Awesome, thanks for the guidance!